tests/sentry/integrations/github/test_integration.py

```diff
@@ -738,3 +738,109 @@ def test_get_trees_for_org_rate_limit_401(self):
                 ("baz", "master", []),
             ]
         )
+
+    @responses.activate
+    def test_get_trees_for_org_makes_API_requests_before_MAX_CONNECTION_ERRORS_is_hit(self):
+        """
+        If some requests fail, but `MAX_CONNECTION_ERRORS` isn't hit, requests will continue
+        to be made to the API.
+        """
+        installation = self.get_installation_helper()
+        self.set_rate_limit()
+
+        # There's no response for repo `xyz` here because the one added in `setUp` is fine
+
+        # Given that below we mock MAX_CONNECTION_ERRORS to be 2, the error we hit here
+        # should NOT force the remaining repos to pull from the cache.
+        responses.replace(
+            responses.GET,
+            f"{self.base_url}/repos/Test-Organization/foo/git/trees/master?recursive=1",
```

Comment:
If make the error happen for `xyz` you won't need to make any changes to the responses for `bar` and `baz`.
You would get the tree for `foo`.